builtin/credential/approle/path_role.go

```diff
@@ -2358,12 +2359,27 @@ func (b *backend) handleRoleSecretIDCommon(ctx context.Context, req *logical.Req
 		return nil, errwrap.Wrapf("failed to store secret_id: {{err}}", err)
 	}
 
-	return &logical.Response{
+	resp := &logical.Response{
 		Data: map[string]interface{}{
 			"secret_id":          secretID,
 			"secret_id_accessor": secretIDStorage.SecretIDAccessor,
 		},
-	}, nil
+	}
+
+	// Follow the behavior from registerSecretIDEntry and set the response TTL
+	// to the entry's TTL unless it's negative or greater than the system's max
+	// lease TTL.
+	switch {
+	case secretIDStorage.SecretIDTTL < time.Duration(0):
+		// This case should not be possible since TypeDurationSecond does not
+		// support negative values, but we follow the same convention as
+		// registerSecretIDEntry
+		resp.Data["ttl"] = int(b.System().MaxLeaseTTL().Seconds())
+	case secretIDStorage.SecretIDTTL > time.Duration(0):
```

Comment:
I wonder if we should always be returning a value, i.e. return `"ttl": 0` on non-expiring secret IDs.

Reply:
Also, should this be `ttl` (conforms to other TTL values on other backends) or `secret_id_ttl` (conforms to the other values in this particular response)?

Reply:
Both comments addressed in 8b90861. @kalafut I've changed it to always return `secret_id_ttl` such that the zero-value indicates non-expiry. This is the same as how other zero-value TTL responses behave throughout Vault.